File: data/readable_format/cpp/file1580.cpp
#include <iostream>
#include <cmath>

const double PI = 3.141593;
const double SOLCON = 0.082;

void radcur(int day, int hr, double rdlat, double decl, double& ratio) {
    double dfact, slrtm, hasun, haset, harise, rdsun;

    dfact = ((2 * PI) * (day - 81)) / 365.0;

    slrtm = (0.1645 * sin(2 * dfact)) - (0.1255 * cos(dfact)) - (0.025 * sin(dfact));

    hasun = ((hr + slrtm) - 12) * (PI / 12);

    haset = hasun - (PI / 24);

    harise = hasun + (PI / 24);

    rdsun = 1 + 0.033 * cos((2 * PI * day) / 365);

    ratio = ((12 * 60) / PI) * SOLCON * rdsun * (cos(rdlat) *
             cos(decl) * (sin(harise) - sin(haset)) +
             (harise - haset) * sin(rdlat) * sin(decl));

    if (ratio < 0.0) {
        ratio = 0.0;
    }
}

void test_radcur(int day, int hr, double rdlat, double decl) {
    double ratio;
    radcur(day, hr, rdlat, decl, ratio);
    std::cout << "Test - Day: " << day << ", Hour: " << hr 
              << ", rdlat: " << rdlat << ", decl: " << decl 
              << ", Ratio: " << ratio << std::endl;
}

int main() {
    // Test Case 1
    test_radcur(100, 12, 0.5, 0.1);
    
    // Test Case 2
    test_radcur(200, 6, 0.5, 0.1);
    
    // Test Case 3
    test_radcur(300, 18, 0.5, 0.1);
    
    return 0;
}

File: data/readable_format/fortran/file1580.f90
program test_radcur
    implicit none

    real :: ratio, rdlat, decl
    integer :: day, hr

    rdlat = 0.5
    decl = 0.1

    ! Test Case 1
    day = 100
    hr = 12
    call radcur(day, hr, rdlat, decl, ratio)
    print *, 'Test 1 - Day:', day, ' Hour:', hr, ' Ratio:', ratio
    
    ! Test Case 2
    day = 200
    hr = 6
    call radcur(day, hr, rdlat, decl, ratio)
    print *, 'Test 2 - Day:', day, ' Hour:', hr, ' Ratio:', ratio
    
    ! Test Case 3
    day = 300
    hr = 18
    call radcur(day, hr, rdlat, decl, ratio)
    print *, 'Test 3 - Day:', day, ' Hour:', hr, ' Ratio:', ratio

contains

    subroutine radcur(day, hr, rdlat, decl, ratio)
        real :: ratio, rdlat, decl
        integer :: day, hr
        real :: pi, slrtm, hasun, haset, harise, solcon, rdsun, dfact

        pi = 3.141593
        solcon = 0.082

        dfact = ((2 * pi) * (day - 81)) / 365

        slrtm = (0.1645 * sin(2 * dfact)) - (0.1255 * cos(dfact)) - &
                (0.025 * sin(dfact))

        hasun = ((hr + slrtm) - 12) * (pi / 12)

        haset = hasun - (pi / 24)

        harise = hasun + (pi / 24)

        rdsun = 1 + 0.033 * cos((2 * pi * day) / 365)

        ratio = ((12 * 60) / pi) * solcon * rdsun * (cos(rdlat) * &
                 cos(decl) * (sin(harise) - sin(haset)) + &
                 (harise - haset) * sin(rdlat) * sin(decl))

        if (ratio < 0.0) then
            ratio = 0.0
        endif
    end subroutine radcur

end program test_radcur